### PR TITLE
add updateBucketById api

### PR DESCRIPTION
**Pull request by nginnever**
This adds an interface to make an update bucket api call to the bridge as request by https://github.com/Storj/storj.js/issues/77

### CODE REVIEW

**Comment by nginnever** [comment]
@retrohacker seems like `getFile` is failing with bad gateway in the tests. Is this a bridge issue you may know of?

File: examples/upload/upload.html [hunk 1]
```diff
@@ -9,21 +9,27 @@
...
       var storj = new Storj(options);
 
-      var rs = storj.download(bucketId, fileId)
+      //var rs = storj.download(bucketId, fileId)
+      // storj.updateBucketById(bucketId, {publicPermissions: ['PULL', 'PULL']}, (err, res) => {
+      //   console.log(err)
+      //   console.log(res)
+      // })
+
+      var rs = {on: function(){}}
```
Comment (retrohacker) [inline examples/upload/upload.html]: I've been seeing code bases use `Function.prototype` for `noop` to keep code coverage at 100%

Thoughts on this pattern? Fine the way it is too.

File: test/all/index.js [hunk 2]
```diff
@@ -208,6 +208,15 @@ test('makePublic', function(t) {
   })
 });
 
+test('updateBucket', function(t) {
```
Comment (retrohacker) [inline test/all/index.js:211]: This shadows the `makePublic` test. Line 220 tests `makePublic` worked by downloading the file using an unauthenticated client. `makePublic` could be broken and this test would 'fix' the remote state by adding `PULL` back to the permissions before we validated we could download the file.

I'm not sure how much we can update on a bucket, but could we change something other than permissions?

Also, we need to not only use `updateBucketById`, but we also need to use `getBucket` after to verify subsequent calls reflect that state.

File: README.md [hunk 3]
```diff
@@ -202,6 +202,25 @@ storj.makePublic(bucketId, ["PULL", "PUSH"], function(e) {
 })
 ```
 
+### `storj.updateBucketById(bucketId, updateInfo, function cb(e) {})`
```
Comment (retrohacker) [inline README.md]: This is nit-picky, but can we change the method to `updateBucket`? We have a pattern going that is `[verb][noun]()` and this breaks that.

Later on we can overload this method to accept names, ids, etc. and derive the type from the form of the first argument.

**Review by retrohacker — CHANGES_REQUESTED** [review]

**Comment by retrohacker** [comment]
Many of these tests build on one another (the current test will make assertions about the state previous tests setup). This is probably a shortcoming in the test suite but seems like the simplicity the pattern affords outweighs the negatives. Happy to be convinced otherwise though :-)

**Comment by retrohacker** [comment]
Taking a look at the tests now

**Comment by retrohacker** [comment]
Its specifically failing on `createToken` and _only_ on an unauthenticated client. Super strange. @phutchins have you seen this?

snapshot1

**Comment by retrohacker** [comment]
It's doing a `POST` to `/buckets/[id]/tokens`

File: examples/upload/upload.html (hunk 1 above)
Comment (nginnever) [reply examples/upload/upload.html]: Ah yeah that would be better but this was just a hack, I have a PR to update and clean up examples that will remove this when this is merged.

File: test/all/index.js (hunk 2 above)
Comment (nginnever) [reply test/all/index.js:211]: Ahh I was afraid of that. We have these options to test updating.

pubkeys - Array
encryptionKey - String
publicPermissions - Array

So i'll just change this to update a pubkey or something that doesn't interfere

File: README.md (hunk 3 above)
Comment (nginnever) [reply README.md]: I am not apposed to that all and like it better myself. The name currently follows the function name used in bridge-client

**Review by retrohacker — APPROVED** [review]

**Comment by RichardLitt** [comment]
👋 Hey! Thanks for this contribution. Apologies for the delay in responding!

We've decided to rearchitect Storj, so that we can scale better. You can read more about this decision [here](https://storj.io/blog/2018/01/getting-from-petabytes-to-exabytes-the-road-ahead/). This means that we are entirely focused on v3 at the moment, in the [storj/storj](https://github.com/storj/storj) repository. Our white paper for v3 is coming **very, very** soon - follow along on [the blog](https://storj.io/blog) and in our [Rocketchat](https://community.storj.io/home).

As this repository is part of the v2 network, we're no longer maintaining this repository. I am going to close this for now. If you have any questions, I encourage you to jump on Rocketchat and ask them there. Thanks!

**Pull request closed without merging**